lib/cacheable/response_cache_handler.rb

```diff
@@ -132,13 +133,18 @@ def serve_from_cache(cache_key_hash, cache_age_tolerance=nil, message = "Cache h
         if cache_age_tolerance && page_too_old(timestamp, cache_age_tolerance)
           Cacheable.log "Found an unversioned cache entry, but it was too old (#{timestamp})"
         else
+          # Bring these instance variables into scope since we execute the block in the scope of the controller.
+          headers = @headers if defined?(@headers)
```

Comment:
should these be passed as an argument to `serve_from_cache` instead? and default to `response.headers`